### PR TITLE
add unknown treatment exception validations

### PR DESCRIPTION
**Pull request by justincorrigible**
Most changes included in the PR were formatting.
The meat of it is adding 'unknown' as an exception for treatments, and ensuring that the fields that shouldn't be present, are indeed not... And plenty of tests added to make sure everything works.

### CODE REVIEW

File: tests/treatment/checkWhenNoTreatment.test.js [hunk 1]
```diff
@@ -26,288 +26,438 @@ const loadObjects = require('../loadObjects');
 // load in all fields with entries prepopulated to null
 const treatment = require('../constructDummyData').getSchemaDummy('treatment');
 
-
 // key -> name of field, value -> unit tests
 const myUnitTests = {
```
Comment (justincorrigible) [inline tests/treatment/checkWhenNoTreatment.test.js:30]: added a whole lot of tests to cover additional edge cases, including handling extended fields, etc.

**Review by justincorrigible — COMMENTED** [review]
The usual comments round to help parse the PR

File: references/validationFunctions/treatment/checkWhenNoTreatment.js [hunk 2]
```diff
@@ -5,42 +5,82 @@
...
+            if (
+                coreFields.includes($name) &&
+                (!$field || $field === null || checkforEmpty($field))
+            ) {
+                return {
+                    valid: false,
+                    message: `The '${$name}' field must be submitted when the 'treatment_type' field is '${treatmentTypes}'`,
+                };
+            } else {
+                // console.log('test passed', $name, $field);
```
Comment (joneubank) [inline references/validationFunctions/treatment/checkWhenNoTreatment.js:69]: unused else blcok should be removed

**Review by joneubank — CHANGES_REQUESTED** [review]

File: references/validationFunctions/treatment/checkWhenNoTreatment.js (hunk 2 above)
Comment (justincorrigible) [reply references/validationFunctions/treatment/checkWhenNoTreatment.js:69]: left it there to indicate meaning/intent. will remove

File: references/validationFunctions/treatment/checkWhenNoTreatment.js [hunk 3]
```diff
@@ -5,42 +5,82 @@
...
+
+        const treatmentExceptionTypes = ['no treatment', 'unknown'];
+
+        // checks for a string just consisting of whitespace
+        const checkforEmpty = entry => {
+            return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'));
+        };
+        const treatmentTypes = $row.treatment_type.map(value => value.toLowerCase());
+
+        const treatmentTypesAreValid = !arrayItemsInSecondArray(
```
[the remark this answers is not in the document]
Comment (joneubank) [reply references/validationFunctions/treatment/checkWhenNoTreatment.js:54]: I think "valid" is the wrong term here. both cases are valid... we have "exception types" though so maybe we could call it `hasExceptionTreatmentType`?

[the remark this answers is not in the document]
Comment (justincorrigible) [reply references/validationFunctions/treatment/checkWhenNoTreatment.js:54]: can do. posting updates

File: references/validationFunctions/treatment/checkWhenNoTreatment.js [hunk 4]
```diff
@@ -5,42 +5,81 @@
...
+
+        const treatmentExceptionTypes = ['no treatment', 'unknown'];
+
+        // checks for a string just consisting of whitespace
+        const checkforEmpty = entry => {
+            return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'));
+        };
+        const treatmentTypes = $row.treatment_type.map(value => value.toLowerCase());
+
+        const recordHasTreatments = !arrayItemsInSecondArray(
```
Comment (justincorrigible) [inline references/validationFunctions/treatment/checkWhenNoTreatment.js:54]: this logic checks the records' treatments, and identifies those that don't have any (upon the logic that "no treatment" and "unknown" are the absence of treatment)

**Review by joneubank — APPROVED** [review]
